contracts/MerkleAirdrop.sol

```diff
@@ -1,51 +1,92 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.8.17;
 
-import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import {MerkleProof} from "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+
 /**
- * @dev A contract allow recipients to claim tokens via merkle airdrop.
+ * @dev A contract that allows recipients to claim tokens via 'merkle airdrop'.
  */
-contract MerkleAirdrop is ERC20 {
-    bytes32 public immutable merkleRoot;
+contract MerkleAirdrop {
+    address public owner;
+    address public sender;
+    IERC20 public token;
+    bytes32 public merkleRoot;
 
-    mapping(address => bool) public hasClaimed; // use uint8
+    mapping(address => uint8) public hasClaimed;
 
-    event Claimed(address indexed to, uint amount);
+    event Claimed(address indexed claimant, uint amount);
 
     /**
      * @dev Constructor
+     */
+    constructor() {
+        owner = msg.sender;
+    }
+
+    /**
+     * @dev initialzer
+     * @param _sender The account from which the tokens are airdropped.
+     * @param _token The address of the airdrop token.
      * @param _merkleRoot The root of the merkle tree contraining the selected recipients
      */
-    constructor(bytes32 _merkleRoot) ERC20("Merkle Airdrop Token", "MAT") {
-        merkleRoot = _merkleRoot;
+    function initialize(
+        address _sender,
+        address _token,
+        bytes32 _merkleRoot
+    ) external onlyOwner {
+        setSender(_sender);
+        setToken(_token);
+        setMerkleRoot(_merkleRoot);
+    }
+
+    modifier onlyOwner() {
+        require(msg.sender == owner, "caller not owner");
+        _;
     }
 
     /**
      * @dev Claims airdropped tokens.
-     * @param to The account being claimed for.
      * @param amount The amount of the claim being made.
      * @param proof A merkle proof proving the claim is valid.
      */
-    // to not required
-    // msg.sender
-    function claim(address to, uint amount, bytes32[] calldata proof) external {
-        require(!hasClaimed[to], "Already claimed");
+    function claim(uint amount, bytes32[] calldata proof) external {
+        require(
+            sender != address(0) &&
+            address(token) != address(0) &&
+            merkleRoot != bytes32(0),
+            "Airdrop is not active"
+        );
```

Comment:
breakdown the require conditions into separate require conditions, instead of appending them with `&&`.